Sender.py

```diff
@@ -0,0 +1,29 @@
+import random 
+import json
+
+def get_temperature_in_C(min_value,max_value,no_of_samples):
+    return generate_samples(min_value,max_value,no_of_samples)
+
+def celcious_to_farenheit_convertor(temp):
+    return round((temp * 1.8) + 32,2)
+     
+def generate_samples(min_value,max_value,no_of_samples):
+    return random.sample(range(min_value, max_value),no_of_samples)
+  
+def get_charge_rate(min_value,max_value,no_of_samples):
+    return generate_samples(min_value,max_value,no_of_samples)
+
+def process_data(temp_in_c, charge_rate):
+    data= {}
+    data.update({"Temperature" : temp_in_c})
+    data.update({"Charge Rate" : charge_rate})
+   
+    return json.dumps(data)
+  
+def display_readings(message):
+    return message
+  
+def formate_message(attribute,readings):
+    return "The {} readings- {}".format(attribute,readings)  
+  
+  
```

Comment:
It would be nice to have a main entry point to Sender. Similar to how you had created for the test file:
if __name__ == '__main__':

This would also allow you to be able to call your 'process_data' function from here and be able to actually simulate/generate the sequence of data required.

Or you can update 'process_data' to actually process and generate the required sequence of data for the sender.